client/src/nat.rs

```diff
@@ -43,7 +43,11 @@ impl<'a> NatTraverse<'a> {
             backend,
             remaining,
         };
-        nat_traverse.refresh_remaining()?;
+
+        // We don't need to reset peer endpoints here as we have not changed their
+        // endpoint to a candidate yet.
+        let _exhausted = nat_traverse.refresh_remaining()?;
```

Comment:
Would it be too hacky to add `peer.endpoint` at the end of `peer.candidates` for each peer with non-empty candidates here?

It has a potential of simiplifiying logic (no need to collect `peers_to_reset`) in this PR but also in the preexisting code: I think that `refresh_remaining()` for example wouldn't have to return `exhaused`.

Reply:
That could work, though the behavior is a bit more implicit. I'll take a look and see how much more straightforward it makes things.

I realized too that people can and will Ctrl-C in the middle of NAT traversal attempts, which could leave peers in a potentially bad state. Should we try to mitigate this, or just expect that the user will `fetch` again if they're having connectivity troubles?

Reply:
> I realized too that people can and will Ctrl-C in the middle of NAT traversal attempts, which could leave peers in a potentially bad state. Should we try to mitigate this, or just expect that the user will fetch again if they're having connectivity troubles?

That would be eventually nice to handle, but maybe postpone for a follow-up PR in order not to complicate this one? Unless you're already into it.

Reply:
@strohel I tried out your suggestion and I like the big simplifications it provided so I pushed a commit with that change.

Reply:
Yay!